File: 523-Continuous Subarray Sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        curr = 0
        hmap = collections.defaultdict(int)
        hmap[0] = -1
        
        for i in range(len(nums)):
            curr = (curr + nums[i])%k
            if curr in hmap:
                if i - hmap[curr] >= 2:
                    return True
            else:
                hmap[curr] = i
        
        return False

File: C++/523-Continuous Subarray Sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> hmap;
        hmap[0] = -1;
        int running = 0, N = nums.size();
        
        for (int i=0; i<N; i++) {
            running += nums[i];
            running%=k;
            if (hmap.contains(running) && i-hmap[running] >= 2) {
                return true;
            }
            if (!hmap.contains(running)) {
                hmap[running] = i;
            }
        }
        return false;
    }
};